src/cluster.c

```diff
@@ -3120,6 +3120,9 @@ void clusterFailoverReplaceYourMaster(void) {
 
     /* 5) If there was a manual failover in progress, clear the state. */
     resetManualFailover();
+    
+    /* 6) initialize vote count */
+    server.cluster->failover_auth_count = 0;
```

Comment:
Doesn't this mean that in the scenario you added (with the double manual failover) the second failover will now wait until the timeout from the initial one before retrying the vote?

I think we need to reset other fields as well (specifically failover_auth_sent to prevent the above - but would be good to clear all the relevant ones to prevent confusion).
I'd create a resetFailoverState function and call it here and in the timeout handling in clusterHandleSlaveFailover

Reply:
I think your suggestion is good，I am trying  to solve this problem completely by adding resetFailoverState function

Reply:
I have add resetFailoverState function when failover replace master  finished and  add it in timeout handling in clusterHandleSlaveFailover

Reply:
> Doesn't this mean that in the scenario you added (with the double manual failover) the second failover will now wait until the timeout from the initial one before retrying the vote?
> 
> I think we need to reset other fields as well (specifically failover_auth_sent to prevent the above - but would be good to clear all the relevant ones to prevent confusion).
> I'd create a resetFailoverState function and call it here and in the timeout handling in clusterHandleSlaveFailover

can you help me check it if it is safe ?